modules/aws_asg/high-cpu-check.bash

```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+set -e
+set -o pipefail
+
+__logger() {
+  level="${1}" && shift
+  msg="${1}" && shift
+  date="$(date --iso-8601=seconds)"
+  log_msg="tag=cron time=${date} level=${level} msg=\"${msg}\" ${*}"
+  logger -t "$(basename "${0}")" "${log_msg}"
+}
+
+__die() {
+  local status="${1}"
+  local code="${2}"
+  __logger "info" \
+    "cron finished" \
+    "status=${status}"
+  exit "${code}"
+}
+
+report_greedy_containers() {
+  max_cpu="${1}"
+  : "${RUNDIR:=/var/tmp/travis-run.d}"
+  instance_id="$(cat "${RUNDIR}/instance-id")"
+  instance_ip="$(cat "${RUNDIR}/instance-ipv4")"
+
+  IFS=$'\n'
+  cpu_usage="$(docker stats --no-stream --format "{{.Container}} {{.CPUPerc}} {{.Name}}" | tr -d '%')"
+
+  if [ -z "${cpu_usage}" ]; then
+    __logger "info" \
+      "no containers running" \
+      "status=noop" \
+      "instance_id=${instance_id}" \
+      "instance_ip=${instance_ip}"
+  fi
+
+  for line in ${cpu_usage}; do
```

Comment:
Can this `for` loop and the zero-result check above be changed into a counter so that only the inner `while read -r ...` is needed and the zero result warning can happen afterward?

Reply:
Due to the fact that modifications to the variable don't persist outside the `while read` loop, I've removed the counter entirely, as I don't think it's really needed anyway.